zebra-state/src/service/finalized_state/zebra_db/transparent.rs

```diff
@@ -438,31 +515,71 @@ impl DiskWriteBatch {
 
                 // Update the address balance by subtracting this UTXO's value, in memory.
                 address_balance_location
-                    .spend_output(&spent_output)
+                    .spend_output(spent_output)
                     .expect("balance underflow already checked");
-                let sending_address_location = address_balance_location.address_location();
 
                 // Delete the link from the AddressLocation to the spent OutputLocation in the database.
                 let address_spent_output = AddressUnspentOutput::new(
                     address_balance_location.address_location(),
-                    spent_output_location,
+                    *spent_output_location,
                 );
                 self.zs_delete(&utxo_loc_by_transparent_addr_loc, address_spent_output);
-
-                // Create a link from the AddressLocation to the spent TransactionLocation in the database.
-                // Unlike the OutputLocation link, this will never be deleted.
-                let address_transaction = AddressTransaction::new(
-                    sending_address_location,
-                    spent_output_location.transaction_location(),
-                );
-                self.zs_insert(&tx_loc_by_transparent_addr_loc, address_transaction, ());
```

Comment:
This is the location of the transaction that created the UTXO, but we need to index the spending transaction.